lib/ActionMetric.js

```diff
@@ -0,0 +1,75 @@
+const { Logger } = require('werelogs');
+
+const MetricsModel = require('./models/MetricsModel');
+
+class ActionMetric {
+    constructor() {
+        this._log = new Logger('ActionMetric');
+    }
+
+    withProducer(producer) {
+        this._producer = producer;
+        return this;
+    }
+
+    withSite(site) {
+        this._site = site;
+        return this;
+    }
+
+    withObjectSize(objectSize) {
+        this._objectSize = objectSize;
+        return this;
+    }
+
+    withEntry(entry) {
+        this._entry = entry;
+        return this;
+    }
+
+    withMetricType(type) {
+        this._metricsType = type;
+        return this;
+    }
+
+    withExtension(extension) {
+        this._extension = extension;
+        return this;
+    }
+
+    _isLifecycleAction() {
```

Comment:
OK for now, I think eventually we should publish all replication metrics while keeping the context info, and do the demultiplexing based on the originator (lifecycle transition etc.) in the higher presentation layers.

Reply:
Ok, I agree with that. This will serve as a kind of placeholder for following-up with a lifecycle metrics feature, and perhaps a service which can present a nice metrics API.